src/main/rust/lib.rs

```diff
@@ -89,6 +77,22 @@ fn print_debug_state() {
     debug!("Release enabled (congrats, this is NOT a debug build).");
 }
 
+#[cfg(feature = "mainnet")]
+pub const NETWORK: MainNetwork = MAIN_NETWORK;
+
+#[cfg(not(feature = "mainnet"))]
+pub const NETWORK: TestNetwork = TEST_NETWORK;
+
+fn wallet_db(env: &JNIEnv<'_>, db_data: JString<'_>) -> Result<WalletDB, failure::Error> {
```

Comment:
no need to change anything (because the goal is to get rid of these DBs on the rust side) but just for future reference:

the convention on the sdk side has been to call the `block_db` here "cache_db" and the `wallet_db` here the "data_db"

Reply:
Yeah, I was aware of the existing terminology; this is kind of a suggestion for how to improve it. The `BlockSource` trait is intended to not be specific to being backed by a database - it's callback-oriented, so it could be backed by a stream, for example.